### ISSUE TITLE
[JENKINS-42364] Editor adaptations for stage based parallel declarative

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/blueocean-plugin#3854]
<p>There is a proposal to clarify how to use parallel with declarative, using stages. This has the benefit of allowing config items that naturally live at stage level to be in parallel branches.</p>

<p>A sample here: <a href="https://gist.github.com/abayer/bbe18e779696af88af9208f7502b898c" class="external-link" target="_blank" rel="nofollow noopener">https://gist.github.com/abayer/bbe18e779696af88af9208f7502b898c</a><br/>
 The declarative ticket for this feature: <a href="https://issues.jenkins-ci.org/browse/JENKINS-41334" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-41334</a><br/>
 PR <a href="https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/152/" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/152/</a></p>

<p>In scope:</p>
<ul>
	<li>Plan what adaptations are required for the new model</li>
	<li>Plan what impl/design is required for editing stage options at the parallel branch level</li>
</ul>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=michaelneale">michaelneale</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-42364" target="_blank">Editor adaptations for stage based parallel declarative</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kzantow.png" title="kzantow's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kzantow">kzantow</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Major
<li><b>component(s)</b>: blueocean-pipeline-editor-plugin, blueocean-plugin
<li><b>resolution</b>: Duplicate
<li><b>resolved</b>: 2017-09-06T18:07:32+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 3
<li><b>imported</b>: 2025-12-01
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>There is a proposal to clarify how to use parallel with declarative, using stages. This has the benefit of allowing config items that naturally live at stage level to be in parallel branches.</p>

<p>A sample here: <a href="https://gist.github.com/abayer/bbe18e779696af88af9208f7502b898c" class="external-link" target="_blank" rel="nofollow noopener">https://gist.github.com/abayer/bbe18e779696af88af9208f7502b898c</a>
 The declarative ticket for this feature: <a href="https://issues.jenkins-ci.org/browse/JENKINS-41334" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-41334</a>
 PR <a href="https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/152/" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/152/</a></p>

<p>In scope:</p>
<ul>
	<li>Plan what adaptations are required for the new model</li>
	<li>Plan what impl/design is required for editing stage options at the parallel branch level</li>
</ul>
</pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> michaelneale:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-42364?focusedId=289675&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-289675">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This could be done as a first pass and have no config, just be an adaptation to the new "model" - and in the second, support branch level config just like at stage level (which I think is not currently supported anyway in the editor)</p>
</pre>
</details>

</ul>
</details>

<p>This could be done as a first pass and have no config, just be an adaptation to the new "model" - and in the second, support branch level config just like at stage level (which I think is not currently supported anyway in the editor)</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> michaelneale:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-42364?focusedId=298100&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-298100">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jamesdumay" class="user-hover" rel="jamesdumay">jamesdumay</a> I think that this one is blocked by the design for setting config on parallel. We could in theory adapt this to work with the new RFE for parallel with no functional change, but that may miss the point (which is that you can have agent config per parallel stage etc). </p>

<p><a href="https://issues.jenkins-ci.org/browse/JENKINS-41541" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-41541</a> is blocking this in that case</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jamesdumay" class="user-hover" rel="jamesdumay">jamesdumay</a> I think that this one is blocked by the design for setting config on parallel. We could in theory adapt this to work with the new RFE for parallel with no functional change, but that may miss the point (which is that you can have agent config per parallel stage etc). </p>

<p><a href="https://issues.jenkins-ci.org/browse/JENKINS-41541" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-41541</a> is blocking this in that case</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kzantow.png" title="kzantow's avatar" /> kzantow:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-42364?focusedId=312836&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-312836">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=michaelneale" class="user-hover" rel="michaelneale">michaelneale</a> I think this has been superseded by it's parts, please reopen if I am missing something.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=michaelneale" class="user-hover" rel="michaelneale">michaelneale</a> I think this has been superseded by it's parts, please reopen if I am missing something.</p>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `blocks` from Jira: <a href="https://github.com/jenkinsci/blueocean-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-41334%22">JENKINS-41334</a>]</i>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `blocks` from Jira: <a href="https://github.com/jenkinsci/blueocean-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-41960%22">JENKINS-41960</a>]</i>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `epic` from Jira: <a href="https://github.com/jenkinsci/blueocean-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-36291%22">JENKINS-36291</a>]</i>
